package.json

```
@@ -32,7 +32,7 @@
     "shared": "yarn workspace @xstate/tools-shared",
     "prepare": "husky install",
     "postinstall": "preconstruct dev && manypkg check",
-    "release": "(cd apps/extension/client && npx vsce package --yarn) && changeset publish && (cd apps/extension/client && npx vsce publish --packagePath $(find . -iname *.vsix))",
+    "release": "(cd apps/extension/client && npx vsce package --yarn) && changeset publish && (cd apps/extension/client && npx vsce publish --packagePath $(find . -iname *.vsix)) && node ./scripts/tag-extension.js",
```

Should this be run after the changeset publish or before? I'm sure you thought of this.

---

it probably doesn't matter - if anything fails during this pipeline then the whole release script should fail and `changesets/action` shouldn't create any github releases (some packages could have been released during the process though, we don't receive "partial" successes anyhow to handle this any better).

The current order matches the behavior of Changesets CLI - it first tries to publish packages to npm and only when it's done with that step then it creates tags for the published packages.

---

OO, one thing we've missed here is swallowing the error from the vsce publish.

```suggestion
    "release": "(cd apps/extension/client && npx vsce package --yarn) && changeset publish && (cd apps/extension/client && npx vsce publish --packagePath $(find . -iname *.vsix)) || true && node ./scripts/tag-extension.js",
```

---

when it can fail? the scenario I imagine in which we should swallow the error is when this particular version has already been published and their store reject us based on that - we shouldn't swallow in any other cases, right?

also - are you sure that this proposed `||` is correct? this probably needs some parens wrapped around this and vsce

---

```suggestion
    "release": "(cd apps/extension/client && npx vsce package --yarn) && changeset publish && (cd apps/extension/client && (npx vsce publish --packagePath $(find . -iname *.vsix)) || true) && node ./scripts/tag-extension.js",
```

---

> Yes, exactly. But how do we selectively swallow that error in bash?

No idea XD I try to stay away from Bash as much as I can :P I guess it's acceptable for now to just roll with this and we see if it ever fails for other reasons

---

Yeah it's much more likely to fail on build, which is a separate proc